client/src/services/currentUser.js

```diff
@@ -51,6 +54,7 @@ export function CurrentUserProvider(props) {
       const { data: payload } = await apiLogin(username);
       if (payload != null) {
         dispatch([types.SUCCESSFUL_GET, payload]);
+        router(`/user/${payload[0].id}/items`);
```

Comment:
as mentioned before this custom hook just adds more complexity. that entire file can be deleted and you can just call `useHistory` in this file and then `history.push` here and on line 42

Reply:
got rid of custom hook